charts/consul/templates/cni-resourcequota.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if .Values.global.imageK8s }}{{ fail "global.imageK8s is not a valid key, use global.imageK8S (note the capital 'S')" }}{{ end -}}
+{{- if (and (.Values.connectInject.cni.enabled) (not .Values.connectInject.transparentProxy.defaultEnabled)) }}{{ fail "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" }}{{ end -}}
+{{- if (and .Values.connectInject.cni.enabled .Values.connectInject.transparentProxy.defaultEnabled) }}
+apiVersion: v1
+kind: ResourceQuota
```

Comment:
We don't define ResourceQuota for any other components. Why do we need this for the CNI?

Reply:
I haven't found an answer for all Kubernetes versions but I do know that GKE wouldn't let me run the daemonset in the kube-system namespace unless I set a ResourceQuota for it.

Reply:
kube-system was a red herring. `system-node-critical` the setting that require resource quotas.

Reply:
ah got it!